File: app/src/main/java/altcoin/br/decred/RestartPriceAlertService.java
package altcoin.br.decred;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import altcoin.br.decred.services.PriceAlertService;

public class RestartPriceAlertService extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Log.e("RestartService", "Restarting PriceAlertService");

        context.startService(new Intent(context.getApplicationContext(), PriceAlertService.class));
    }
}

File: app/src/main/java/altcoin/br/decred/RestartPriceAlertService.kt
package altcoin.br.decred

import altcoin.br.decred.services.PriceAlertService
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class RestartPriceAlertService : BroadcastReceiver() {
	override fun onReceive(context: Context, intent: Intent) {
		Log.e("RestartService", "Restarting PriceAlertService")

		context.startService(Intent(context.applicationContext, PriceAlertService::class.java))
	}
}
